app/experimenter/experiments/views.py

```diff
@@ -279,6 +288,25 @@ def subscribed_filter(self, queryset, name, value):
 
         return queryset
 
+    def longrunning_filter(self, queryset, name, value):
+        if value:
+            return (
+                queryset.exclude(firefox_max_version__exact="")
+                .annotate(
+                    firefox_min_float=Cast(
+                        "firefox_min_version", FloatField()
+                    ),
+                    firefox_max_float=Cast(
+                        "firefox_max_version", FloatField()
+                    ),
+                    version_count=F("firefox_max_float")
+                    - F("firefox_min_float"),
+                )
+                .filter(version_count__gte=3.0)
```

Comment:
This is clever and elegant and works well for the way we have the version values setup (they easily parse into floats), but I wonder what would happen if we added some versions that contain non numerical characters.  It hasn't happened yet but in theory it could.  Doing it this way could introduce a little hidden contact that the version value strings must be parseable into floats otherwise this breaks.  What do you think?

Reply:
What non-numerical versions are you thinking might happen?

Reply:
Like if we add some special version strings in the future that might look like "62.01a" or include some non numerical characters, or if we add the ability for people to specify custom versions in the future.  Doesn't necessarily mean we have to deal with a potentially non existent future case right now, but just to think about if we do in the future, and if this breaks, will it be clear what went wrong or what the baked in assumptions are?

Reply:
Hmm okay, well yes I feel like it would be clear that if you made a significant change to the way we represent versions that it would likely break the filters that have to parse and do math to versions. Is there something you were thinking I could add that would make you feel more confident about this approach? I think if I wanted to build a longrunning filter right now that allowed versions with nonnumerical characters, I'd have to do some raw sql stuff instead of using the ORM methods, and nobody seems to like that idea generally.

Reply:
http://andilabs.github.io/django/django-1-8/expressions/func/f/2015/08/29/django-1.8-new-db-models-expressions-F-Func.html

Reply:
@uhlissuh Oh the link was here!

Reply:
@jaredkerim OMGGGG.